lucene/core/src/java/org/apache/lucene/util/fst/FSTEnum.java

```diff
@@ -178,7 +179,7 @@ protected void doSeekCeil() throws IOException {
     } else {
       if (targetIndex < 0) {
         targetIndex = -1;
-      } else if (arc.bitTable().isBitSet(targetIndex)) {
```

Comment:
This is just a side note but it's code like this one that would really benefit from some kind of arc iterator abstraction. :)

Reply:
I agree. I'd like to review FSTEnum soon. First to refactor and share common code like this, second there is still room for some perf improvement I think for seek floor/ceil.